make unit_test fails with module not found error (regression)

/kind bug

**What steps did you take and what happened:**

```Bash
$ make unit_test
==================================================================
Optional environment variables to configure unit_test, examples:
  make unit_test GENERATE_GOLDEN_YAML=True
==================================================================
compiler_tests (unittest.loader._FailedTest) ... ERROR
test_convert_k8s_obj_to_json_accepts_dict (compiler.k8s_helper_tests.TestK8sHelper) ... ok
test_sanitize_k8s_annotations (compiler.k8s_helper_tests.TestK8sHelper) ... ok
test_sanitize_k8s_labels (compiler.k8s_helper_tests.TestK8sHelper) ... ok
test_sanitize_k8s_name_max_length (compiler.k8s_helper_tests.TestK8sHelper) ... ok

======================================================================
ERROR: compiler_tests (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: compiler_tests
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/Users/ckadner/PycharmProjects/kfp-tekton_ckadner/sdk/python/tests/compiler/compiler_tests.py", line 25, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'


----------------------------------------------------------------------
Ran 5 tests in 0.001s

FAILED (errors=1)
make: *** [unit_test] Error 1
```

**What did you expect to happen:**

`Makefile` to ensure `pytest` exists or gets installed into the `make` process' `virtualenv`.

**Additional information:**

Regression introduced with #283


**Environment:**

* Python Version (use `python --version`): `2.7.5`
* SDK Version: `master`
* Tekton Version (use `tkn version`): n/a
* Kubernetes Version (use `kubectl version`): n/a
* OS (e.g. from `/etc/os-release`): Mac OS

/assign

--- a/.travis.yml
+++ b/.travis.yml
@@ -20,7 +20,6 @@ matrix:
       env: TOXENV=py35
       install: &0
         - python3 -m pip install -e sdk/python
-        - python3 -m pip install pytest
       script: &1
         - make unit_test
     - name: "Unit tests, Python 3.6"

--- a/Makefile
+++ b/Makefile
@@ -42,6 +42,7 @@ unit_test: venv ## Run compiler unit tests
 	@echo "Optional environment variables to configure $@, examples:"
 	@sed -n -e 's/# *\(make $@ .*\)/  \1/p' sdk/python/tests/compiler/compiler_tests.py
 	@echo "=================================================================="
+	@pip show pytest > /dev/null 2>&1 || pip install pytest
 	@sdk/python/tests/run_tests.sh
 	@echo "$@: OK"
 

